Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Finding,
+  HandleBlock,
+  getEthersProvider,
+  BlockEvent,
+} from "forta-agent";
+import {
+  DAI_ADDRESS,
+  L1_ESCROW_ARBITRUM,
+  L1_ESCROW_FUNCTION_SIGNATURE,
+  L1_ESCROW_OPTIMISM,
+} from "./constants";
+import { ethers, Contract } from "ethers";
+import { getL1Finding, checkBlock } from "./utils";
+
+export function provideHandleBlock(
+  provider: ethers.providers.Provider,
+  alert: any,
+): HandleBlock {
+  return async function handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    const { chainId } = await provider.getNetwork();
+
+    try {
+      if (chainId == 1) {
+        const daiContract = new Contract(
+          DAI_ADDRESS,
+          L1_ESCROW_FUNCTION_SIGNATURE,
+          provider,
+        );
+        findings.push(
+          await getL1Finding(
+            daiContract,
+            block.blockNumber,
+            L1_ESCROW_ARBITRUM,
+            L1_ESCROW_OPTIMISM,
+          ),
+        );
+      } else {
+        const blockFindings = await checkBlock(
+          provider,
+          block.blockNumber,
+          chainId,
+          alert,
+        );
+
+        if (blockFindings.length > 0) {
+          findings.push(...blockFindings);
+        }
```

Comment:
Although you correctly split the logic based on the chain ID, this can't work as the `alert` that you pass to `checkBlock` will always be empty.

(Each bot instance is run either on L1 or L2 based on what the Forta node chooses - L1 and L2 instances don't communicate directly with each other)

Reply:
Hey Vasilis, thanks for the feedback!

What is a better approach for passing in alert to checkBlock? My approach was to pass it in as a parameter but I wanted to know what a better approach for passing in/raising an alert is in general.

Reply:
In this case, one of the two sides (L1 or L2) needs to listen to the alerts of the other (inside the handleBlock). There is an SDK method to achieve this (supposing that the bot is indeed deployed, and we know its ID)

Reply:
What would be the best approach to achieve this? If the sdk method for this is unusable if the bot isn't deployed.

Would a possible approach be to set up a listener/read in for alerts in the handleblock of the L2 side as you suggested above?

Reply:
Hey Vasilis, I added a listener from the L2 side to listen for L1 alerts, let me know if my implementation of this is correct, thanks!